ctapipe/image/extractor.py

```diff
@@ -1190,23 +1190,23 @@ def __call__(self, waveforms, telid, selected_gain_channel):
             Shape: (n_pix)
         """
 
-        self._charge1, self._pulse_time1, correction1 = self._apply_first_pass(
+        charge1, pulse_time1, correction1 = self._apply_first_pass(
             waveforms, telid
         )
 
         # FIXME: properly make sure that output is 32Bit instead of downcasting here
         if self.disable_second_pass:
             return (
-                (self.charge1 * correction1[selected_gain_channel]).astype("float32"),
-                self._pulse_time1.astype("float32"),
+                (charge1 * correction1[selected_gain_channel]).astype("float32"),
```

Comment:
the typo was here I guess, since self.charge1 was used instead of self._charge1.  The reason for making those class variables (rather than local ones) was to  support being able to debug the first and second passes without having to re-compile or re-run the code.  Is that no longer needed? I.e if you want to check the output of the first pass only, you need these.

Reply:
No, since you can deactivate the second pass. So you can just deactivate the second pass to check only the first pass.

Reply:
Or you just directly call the `first_pass` and `second_pass` methods.

Reply:
this is what I did now: I added a test where I disable 2nd pass (which is indeed a traitlet by itself, so I guess we debug 1st pass by disablng the 2nd)